src/actions/actions.tsx

```diff
@@ -0,0 +1,25 @@
+import { History, LocationState } from 'history';
+import { auth, provider } from '../firebase';
+
+interface SignInWithPwdEmailProps {
+  email: string;
+  password: string;
+  history: History<LocationState>;
+}
+interface SignInWithGoogle {
+  history: History<LocationState>;
+}
+
+export const signInWithPasswordAndEmail = ({ email, password, history }: SignInWithPwdEmailProps) => {
```

Comment:
you could split signInWithPasswordAndEmail and signInWithPopup to two different files

Reply:
SignInWithPwdEmailProps it's not React component so it shouldn't be name Props. It could SignInWithPwdEmail